circle.yml

```diff
@@ -9,18 +9,37 @@ workflows:
   default:
     jobs:
       - build
+      - static-analysis:
+          requires:
+            - build
+      - unit-tests:
+          requires:
+            - build
+      - ui-robo-tests:
+          requires:
+            - build
```

Comment:
This job runs `release` variants of libraries while `build` builds `debug` variants, so `ui-robo-tests` doesn't have to rely on `build`. How about we make `build` job a `build-debug` and add another `build-release` job that `ui-robo-tests`, `release-snapshot` and `release` can require?

Reply:
`build` combines all variants(debug+release), but I think we could replace it by `assemble`(that is assemble for _Release_ and _Debug_ at the same time), wdyt?

Reply:
As far as I can see, `build` only generates debug artifacts - https://github.com/mapbox/mapbox-navigation-android/blob/e5ce9590acbd2ac7e2b079ac553af975fd98c561/circle.yml#L306-L314

Reply:
@LukasPaczos nice note, I didn't expect that "build" is actually "assemble"
https://github.com/mapbox/mapbox-navigation-android/blob/e7be76c5af257195c87bef420be455d6a136fc96/Makefile#L66-L72
and 
https://github.com/mapbox/mapbox-navigation-android/blob/e7be76c5af257195c87bef420be455d6a136fc96/Makefile#L106-L112
what do you think if we rename these tasks at `make` file and rename tasks at `CI` config

**cc** @Guardiola31337

Reply:
I don't mind, whatever works best :)

Reply:
@LukasPaczos have suggestion: let's not run `firebase` tests(robo and instrumentation) if we don't archive all previous checks(assemble release+debug, unit tests, lint, docs, check api). 
Pros: 
- saves money(firebase is not free) 
- keeps more machines ready for other builds(we have limited range of machines at circle ci, correct me if I'm wrong)
- CI runs firebase tests anyway, when previous fails are solved
Cons:
- increase time of tests(ui and robo tests starts after all previous checks passed)

Reply:
Sounds good @RingerJK :+1: 

@mapbox/navigation-android let's monitor the duration of our CI jobs and if they start getting out of hand, we can revisit this setup. For code-unrelated changes, we can work on https://github.com/mapbox/mapbox-navigation-android/issues/3544.